src/metadata.rs

```diff
@@ -158,13 +212,37 @@ impl FromStr for Metadata {
             &mut title_end,
             |caps| {
                 caps.name("short")
-                    .or_else(|| caps.name("long"))
                     .or_else(|| caps.name("cross"))
                     .or_else(|| caps.name("dash"))
                     .map(|m| m.as_str())
             },
         );
-
+        let episode_num = episode.map(|s| s.parse().unwrap());
+        // Only look for a last episode if one was found prevkously.
+        let mut last_episode_num: Option<i32> = None;
+        if let Some(first_episode) = episode {
+            last_episode = check_pattern_and_extract(
+                &pattern::LAST_EPISODE,
+                name,
+                &mut title_start,
+                &mut title_end,
+                |caps| caps.get(1).map(|m| m.as_str()),
+            );
+            // Sanity check that last_episode does not contain a value or 0 (Zero)
+            if let Some(l_episode) = last_episode {
+                if l_episode.len() == 1 && l_episode.contains('0') {
+                    last_episode = None;
+                } else {
+                    // Populate multi_episode
+                    let mut episode_numbers: Vec<i32> = Vec::new();
+                    for i in first_episode.parse().unwrap()..=l_episode.parse().unwrap() {
+                        episode_numbers.push(i);
+                    }
+                    multi_episode = Some(episode_numbers);
+                }
+            }
+            last_episode_num = last_episode.map(|s| s.parse().unwrap());
+        }
```

Comment:
The logic looks correct to me, but the flow and naming could be much clearer. this should probably be a single functional instruction that eventually resolves to an Option<i32>, not doing a bunch of conditional assigning.

Reply:
I have made some improvements. At this stage I still need some conditionals. 

Will submit when I have some documentation done as well.

Reply:
Submitted changes